src/commands/blocking_commander.h

```diff
@@ -44,6 +47,10 @@ class BlockingCommander : public Commander,
   // in other words, returning true indicates ending the blocking
   virtual bool OnBlockingWrite() = 0;
 
+  // method to get keys to be locked
+  // when OnWrite is triggered, BlockingCommander needs to relock the keys
+  virtual std::vector<std::string> GetLockKeys() = 0;
```

Comment:
I'm wonder if it's better to be `MultiLockGuard GetLock()` rather than returning a vector?